persist-cli/src/main/java/io/ballerina/persist/nodegenerator/syntax/utils/BalSyntaxUtils.java

```diff
@@ -406,6 +416,65 @@ private static ModuleMemberDeclarationNode createEntityRecordOptionalized(Entity
                 entity.getEntityName().trim(), recordFields));
     }
 
+    private static void addConstraintsAnnotationForForeignKey(EntityField field, StringBuilder recordFields) {
+        Relation relation = field.getRelation();
+        List<String> references = relation.getReferences();
+        for (EntityField assocField : relation.getAssocEntity().getFields()) {
+            for (String reference : references) {
+                if (assocField.getFieldName().equals(reference)) {
+                    NodeList<AnnotationNode> annotation = assocField.getAnnotation();
+                    if (annotation != null) {
```

Comment:
If it's possible, it's better to check if the containing values in the annotation are constraint type right?

Reply:
I already checked in. https://github.com/ballerina-platform/persist-tools/pull/205/files/cfc1c1149b5ae542c1138a4ba992264dfdc27f4f#diff-0aa938f8541470721c9aa46c3dfe249392b067e7563ede93708a622d62d7d874R452